### PR TITLE
added spark 2.2.0 build support

### PR DESCRIPTION
**Pull request by cseed**
passes all tests, builds package locally
ripped out (now unused) mongo dependencies

### CODE REVIEW

**Comment by cseed** [comment]
Also removed 2.1.0 from deployed versions.  When this goes in I'll remove the CI tests againt 2.1.0.

File: build.gradle [hunk 1]
```diff
@@ -34,29 +34,39 @@ buildScan {
 }
 
 String hailVersion = "devel"
-String[] deployedSparkVersions = ["2.0.2", "2.1.0"]
+String[] deployedSparkVersions = ["2.0.2"]
```
Comment (tpoterba) [inline build.gradle]: should we add 2.2.0 to the deployed versions?

File: build.gradle [hunk 2]
```diff
@@ -212,7 +220,7 @@ task testPython(type: Exec, dependsOn: shadowJar) {
      commandLine 'python', '-m', 'unittest', 'hail.api1.tests', 'hail.api2.tests', 'hail.typecheck.tests', 'hail.genetics.tests', 'hail.expr.tests', 'hail.utils.tests', 'hail.methods.tests'
      environment SPARK_HOME: sparkHome
      environment PYTHONPATH: '' + projectDir + '/python:' + sparkHome + '/python:' + sparkHome + '/python/lib/py4j-' + py4jVersion + '-src.zip'
-     environment SPARK_CLASSPATH: '' + projectDir + '/build/libs/hail-all-spark.jar'
+     environment PYSPARK_SUBMIT_ARGS: '--conf spark.driver.extraClassPath=' + projectDir + '/build/libs/hail-all-spark.jar --conf spark.executor.extraClassPath=' + projectDir + '/build/libs/hail-all-spark.jar pyspark-shell'
```
Comment (tpoterba) [inline build.gradle]: what is the `pyspark-shell` doing here?

File: build.gradle [hunk 3]
```diff
@@ -408,14 +410,14 @@ task makeFunctionsRst(type: Exec, dependsOn: ['shadowJar', 'copyPythonDirToTmp']
     commandLine 'python', '-c', 'from hail import *; from hail.utils import *; hc = HailContext(); fd = FunctionDocumentation(); fd.types_rst("python/hail/docs/types.rst"); fd.functions_rst("python/hail/docs/functions.rst")'
     environment PYTHONPATH: '' + projectDir + '/python:' + sparkHome + '/python:' + sparkHome + '/python/lib/py4j-' + py4jVersion + '-src.zip'
     environment SPARK_HOME: sparkHome
-    environment SPARK_CLASSPATH: '' + projectDir + '/build/libs/hail-all-spark.jar'
+    environment PYSPARK_SUBMIT_ARGS: '--conf spark.driver.extraClassPath=' + projectDir + '/build/libs/hail-all-spark.jar --conf spark.executor.extraClassPath=' + projectDir + '/build/libs/hail-all-spark.jar pyspark-shell'
```
Comment (tpoterba) [inline build.gradle]: same here

File: build.gradle (hunk 1 above)
Comment (cseed) [reply build.gradle]: I was going to do that after 2.2.0 was being tested so we're not deploying something (too) broken.

File: build.gradle (hunk 2 above)
Comment (cseed) [reply build.gradle]: See here for example: https://issues.apache.org/jira/browse/SPARK-17387

And the comment towards the bottom: "BTW my workaround above is wrong - "pyspark-shell" needs to be the last argument in "PYSPARK_SUBMIT_ARGS", not the first."

Comment (tpoterba) [reply build.gradle]: This is super janky and I don't like it, but you've convinced me it's right.

File: build.gradle (hunk 1 above)
Comment (tpoterba) [reply build.gradle]: ah, sure.

**Comment by cseed** [comment]
So overall plan is:

 - get 2.2.0  build support in (this patch)
 - stop testing 2.1.0 once it isn't being deployed,
 - start testing 2.2.0 (I will need to update the CI image to install Spark 2.2.0),
 - add 2.2.0 to the list of deployed versions,
 - make Spark 2.2.0/Dataproc 1.2 the version in cloudtools,
 - drop testing/deploy support for 2.0.2

Did I miss anything?

**Comment by tpoterba** [comment]
That sounds complete. Annoying that we need to change 3 independent pieces!

**Pull request merged**